examples/with-style-sheet/next.config.js

```diff
@@ -3,9 +3,8 @@ const { RawSource } = require('webpack-sources')
 
 class StyleSheetPlugin {
   apply(compiler) {
-    compiler.plugin('emit', (compilation, cb) => {
+    compiler.hooks.emit.tap('StyleSheetPlugin', (compilation) => {
       compilation.assets['static/bundle.css'] = new RawSource(getCss())
```

Comment:
I didn't have time to look into this but ideally I'd push this asset to the array of CSS files and let Next.js render it in the head. This way we could add a hash to the file. Please let me know in case this is a quick fix otherwise I'll contribute with that another time.

Reply:
@Timer I noticed that upon restarting the dev server this file is often empty. 

I assume that the reason for that is that the webpack plugin runs again but perhaps the build is cached and the Babel plugin won't collect the styles again -> getCss() is flushing an empty style sheet.

Has anybody else had this type of issue before? I figured I'd ask before I start to dig/debug this issue.

Reply:
Ah yes, you cannot rely on state from the babel plugins because it'll use the cache.

Reply:
Right, well I guess the solution would be to cache the css file then. I'll look into that when I have a bit of time, thanks!

Reply:
@Timer would it make sense to extract the styles (run the babel plugin) only when building for prod? I think that'd work but I'd like to know how to cache an emitted asset nonetheless (assuming that's doable)